_projects/coala_language_server.md

```diff
@@ -42,48 +42,110 @@ a rapidly growing list of editor and language communities:
 We have a state-of-art implementation in
 [coala-vs-code](https://github.com/coala/coala-vs-code) but it is designed
 for VS Code only. We need to design and implement a general mechanism for
-multiple editors/IDEs support.
+multiple editors/IDEs support. We also need clients written to support coala
+langserver for most if not all the supported editors. This is not because of
+the server's inability, but rather the need to configure the front end of the
+editor plugin such as defining how to start the server, communicate with it etc.
 
-The project is divided into two parts:
+The project is divided into three parts:
 
-1. Offer a basic languager server.
+1. Offer a basic language server.
 2. Implement a plugin based on the language server.
+3. Optimize the coala entry point and language server.
 
 #### Features
 
 - Support linting for multiple programming languages.
-- Support 'textDocument/publishDiagnostics', 'textDocument/didChange' and 'textDocument/didSave' requests.
+- The current language server has implementations for:
+  - 'textDocument/didSave'
+  - 'textDocument/didChange'
+  - 'textDocument/publishDiagnostics'
+  and this project will improve and extend the support to at least
+  the following request types:
+  - 'textDocument/didOpen'
+  - 'textDocument/didClose'
+  - 'workspace/formatting'
+  - 'window/showMessageRequest'
 - (Advanced) Multiple plugins Support (e.g. Eclipse Che).
+- Some additional request types that are either out of the scope
+  of this project or are marginally redundant to build but which would
+  improve the user friendliness or functionality  of this language
+  server are:
+  - 'textDocument/willSaveWaitUntil'
+  - 'workspace/didChangeConfiguration'
+  - 'workspace/configuration'
+  - 'workspace/applyEdit'
+  - 'window/showMessage'
+
+#### Interface Optimizations
+
+The current implementation suffers from massive performance
+problems. This is primarily due to the mode in which lang
+server interacts with coala. Currently it works by executing
+coala via cli and redirecting stdout to language server. This
+is not an efficient mechanism because of all the involved overhead.
+This projects aims to provide a practical language server with
+performance improvements. Some ideas currently under consideration
+to achieve the said improvements are:
+
+  - Runtime coala configurations
+  - Language Server specific parameters in coafiles
+  - Custom entry point into coala
+
+Specific details about each method will be expanded upon
+once they are discussed with the community.
 
 #### Milestones
 
 ##### PREPARATION/BONDING
 
 - Design documentation about the language server exists.
-  - Architecture of the language server
-  - Mechanism to support multiple languages.
+  - Architecture of the language server.
+  - Mechanism to support multiple languages and requests.
```

Comment:
I would suggest one language server per language, instead of trying to make one language server support multiple languages. We've tried it, it doesn't work so well!

Reply:
It may well be the same code runs with different config for each language.

Reply:
Yes, support for multiple languages is more about configuring the clients to handle multiple file types. coala Language Server only depends on the coafile configuration to handle it. I do not think there should be any significant problem with the server.